src/lib/utils/meta_tables/meta_system_information_table.cpp

```diff
@@ -0,0 +1,89 @@
+#ifdef __linux__
+#include <numa.h>
+#include <sys/sysinfo.h>
+#endif
+
+#include <fstream>
+
+#include "meta_system_information_table.hpp"
+
+namespace opossum {
+
+MetaSystemInformationTable::MetaSystemInformationTable()
+    : AbstractMetaSystemTable(TableColumnDefinitions{{"cpu_count", DataType::Int, false},
+                                                     {"system_memory_total_bytes", DataType::Long, false},
+                                                     {"numa_cpu_count", DataType::Int, false},
+                                                     {"cpu_model", DataType::String, false}}) {}
+
+const std::string& MetaSystemInformationTable::name() const {
+  static const auto name = std::string{"system_information"};
+  return name;
+}
+
+std::shared_ptr<Table> MetaSystemInformationTable::_on_generate() const {
+  auto output_table = std::make_shared<Table>(_column_definitions, TableType::Data, std::nullopt, UseMvcc::Yes);
+
+  const auto cpus = _get_cpu_count();
+
+  auto numa_cpus = cpus;
+#ifdef __linux__
+  if (numa_available() != -1) {
+    auto* cpu_mask = numa_allocate_cpumask();

```

Comment:
I believe libnuma is optional. Either check HYRISE_NUMA_SUPPORT or make it required (and update CMake accordingly).

Reply:
Looking at this again, CPU affinities can also be present without libnuma installed. https://linux.die.net/man/2/sched_getaffinity should work just as well.

Did you check if OS X has scheduling affinities? If it doesn't, we should doc this behavior here.

Reply:
this is now solved kind of indirect. Topology cares about HYRISE_NUMA_SUPPORT and "fakes" the numa topology if required.